File: SDKs/soundcloud.com/1.0.0/go/pkg/models/operations/getmelikestracks.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetMeLikesTracksQueryParams struct {
	Limit              *int64 `queryParam:"style=form,explode=true,name=limit"`
	LinkedPartitioning *bool  `queryParam:"style=form,explode=true,name=linked_partitioning"`
}

type GetMeLikesTracksSecurity struct {
	AuthHeader shared.SchemeAuthHeader `security:"scheme,type=apiKey,subtype=header"`
}

type GetMeLikesTracksRequest struct {
	QueryParams GetMeLikesTracksQueryParams
	Security    GetMeLikesTracksSecurity
}

type GetMeLikesTracksResponse struct {
	ContentType                             string
	Error                                   *shared.Error
	StatusCode                              int64
	GetMeLikesTracks200ApplicationJSONOneOf *interface{}
}

File: SDKs/soundcloud.com/1.0.0/typescript/src/sdk/models/operations/getmelikestracks.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetMeLikesTracksQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=linked_partitioning" })
  linkedPartitioning?: boolean;
}


export class GetMeLikesTracksSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  authHeader: shared.SchemeAuthHeader;
}


export class GetMeLikesTracksRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetMeLikesTracksQueryParams;

  @SpeakeasyMetadata()
  security: GetMeLikesTracksSecurity;
}


export class GetMeLikesTracksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getMeLikesTracks200ApplicationJSONOneOf?: any;
}
